pyriemann/transfer/_euclidean.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+
+from scipy.linalg import sqrtm, inv
+from numpy import iscomplexobj, real, any, isfinite
+from sklearn.base import BaseEstimator, TransformerMixin
+from pyriemann.estimation import covariances
+
+from pyriemann.utils.mean import mean_covariance
+from pyriemann.transfer import decode_domains
+
+
+def compute_ref_euclidean(data):
+    '''
+    Calculate the reference matrix for computing the Euclidean alignment,
+
+    Parameters
+    ----------
+    data: numpy array.
+    It can be either raw trials with shape (n, c, t) or covariances (n, c, c)
+    Reference trials to compute the alignment matrix
+
+    Returns
+    -------
+    r_ea: numpy array with shape (c, c)
+    Reference matrix
+
+    '''
+
+    mean = mean_covariance(data, metric='euclid')
+
+    compare = np.allclose(mean, np.identity(mean.shape[0]))
+
+    if not compare:
+        if iscomplexobj(mean):
+            print("covariance matrix problem")
+        if iscomplexobj(sqrtm(mean)):
+            print("covariance matrix problem sqrt")
+
+        r_ea = inv(sqrtm(mean))
+
+        if iscomplexobj(r_ea):
+            print("WARNING! Covariance matrix was not SPD somehow. " +
+                  "Can be caused by running ICA-EOG rejection, if " +
+                  "not, check data!!")
+            r_ea = real(r_ea).astype(np.float64)
+        elif not any(isfinite(r_ea)):
+            print("WARNING! Not finite values in R Matrix")
+
+    else:
+        print("Already aligned!")
+        r_ea = mean
+
+    return r_ea
+
+
+class TransformEA(BaseEstimator, TransformerMixin):
+    """Euclidean Alignment, based on [1].
+
+    The Euclidean Alignment is a transformation that recenter each domain's
+    Euclidean mean covariance matrix to the Identity, making distributions
+    more similar.
+
+    This transformation accept both raw signals and covariance matrices as
+    data inputs. For raw trials, you can determine the estimator using the
+    'estimtor' parameter.
+
+    .. note::
+       Using .fit() and then .transform() will give different results than
+       .fit_transform(). In fact, .fit_transform() should be applied on the
+       training dataset (target and source) and .transform() on the test
+       partition of the target dataset.
```

Comment:
I think this is the default behavior for transformer_mixin.
https://github.com/scikit-learn/scikit-learn/blob/f07e0138bfee41cd2c0a5d0251dc3fe03e6e1084/sklearn/base.py#L1043

Reply:
You might be right. I was not sure if it was the case, so I put this comment. Since I was following [https://github.com/pyRiemann/pyRiemann/blob/master/pyriemann/transfer/_estimators.py#L89](https://github.com/pyRiemann/pyRiemann/blob/c44b924573dc2f4b4a9efb328d18636b8baa31d5/pyriemann/transfer/_estimators.py#L89) as a reference to my code, I put the same note.